### ISSUE TITLE
Check function calling inside functions using `custom_rules`

### ISSUE THREAD

**Issue by maximbilan** [realm/SwiftLint#3500]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I use `custom_rule` to show warnings for all `@IBAction` functions which don't call `handleActionButton` function inside these functions and it doesn't work correctly.
In that case, `SwiftLint` doesn't work correctly.

Here is the custom rule:
```yaml
custom_rules:
  button_action_tracking:
    name: "Button Action Tracking"
    regex: "@IBAction\\s+func\\s.*\\s(?!.*handleButtonAction)\\s"
    message: "Don't forget to call `handleButtonAction` inside the method"
    severity: warning
```

Input:
```swift
class Test {
    
    @IBOutlet weak var button1: UIButton!
    @IBOutlet weak var button2: UIButton!
    @IBOutlet weak var button3: UIButton!

    func handleButtonAction(with button: UIButton) {
    }

    @IBAction func button1Tapped() {
        handleButtonAction(with: button2)
    }

    @IBAction func button2Tapped() {
    }

    @IBAction func button3Tapped() {
    }

}
```

Linting:
```bash
$ swiftlint lint --path file1.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at path file1.swift
Linting 'file.swift' (1/1)
file1.swift:10:5: warning: Button Action Tracking Violation: Don't forget to call `handleButtonAction` inside the method (button_action_tracking)
file1.swift:20:1: warning: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
file1.swift:2:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
Done linting! Found 3 violations, 0 serious in 1 file.
```

However, `SwiftLint` highlights the first function instead of the second and third.

### Expected behavior

https://regex101.com/r/TdkiNC/32

![Screen Shot 2021-01-22 at 10 52 43](https://user-images.githubusercontent.com/3020089/105468802-0d254080-5ca0-11eb-948a-fcb3b7909791.png)

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.42.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
CocoaPods
* Configuration file:

```yml
excluded:
  - Pods
  - vendor

disabled_rules:
  - line_length
  - identifier_name
  - function_parameter_count

force_cast: warning
force_try: warning
cyclomatic_complexity:
  error: 40
type_body_length:
  error: 1000
function_body_length:
  error: 200
file_length:
  error: 1600

custom_rules:
  button_action_tracking:
    name: "Button Action Tracking"
    regex: "@IBAction\\s+func\\s.*\\s(?!.*handleButtonAction)\\s"
    message: "Don't forget to call `handleButtonAction` inside the method"
    severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No
* Which Xcode version are you using (check `xcodebuild -version`)?
12.3 (12C33)

**Comment by coolbnjmn** [comment]
For what it's worth, this looks related to a couple issues that were never resolved and closed for staleness:
https://github.com/realm/SwiftLint/issues/2677
https://github.com/realm/SwiftLint/issues/2040

**Comment by vladimir-bebeshko** [comment]
are there plans to fix this?

**Comment by stonko1994** [comment]
I had a similar issue and after some digging into `SwiftLint` I found out that the problem is the `new-line` matching.

When a custom rule is applied, `SwiftLint` passes the whole content of the file as a single-line to the `NSRegularExpression`.

See: https://github.com/realm/SwiftLint/blob/2408839be3b312bfbbb81811f2fde29d2d485fac/Source/SwiftLintFramework/Extensions/SwiftLintFile%2BRegex.swift#L107-L110

- `stringView` comes from `SourceKitten` and `stringView.string` returns the file content as a single line where new-line characters are replaced with `\n`.

---

I made my regex work after changing the regex to also work with a single line input.
